docs/dev/replay-e2e-test.md

```diff
@@ -0,0 +1,51 @@
+# Replay End-to-End Tests
+
+* This document introduces replay tests for those located in [sdk_cli_azure_test](../../src/promptflow/tests/sdk_cli_azure_test/e2etests/) and [sdk_cli_test](../../src/promptflow/tests/sdk_cli_test/e2etests/).
+* The primary purpose of replay tests is to avoid the need for credentials, Azure workspaces, OpenAI tokens, and to directly test PromptFlow behavior.
+* Although there are different techniques behind recording/replaying, there are some common steps to run the tests in replay mode.
+* The key handle of replay tests is the environment variable `PROMPT_FLOW_TEST_MODE`.
+
+## (1 Step Version) How to Run Tests in Replay Mode
+
+After cloning the full repo and setting up the proper test environment following [dev_setup.md](./dev_setup.md), run the following command in the root directory of the repo:
+
+1. If you have changed/affected tests in __sdk_cli_test__ : Copy or rename the file [dev-connections.json.example](../../src/promptflow/dev-connections.json.example) to `connections.json` in the same folder.
+2. In your Python environment, set the environment variable `PROMPT_FLOW_TEST_MODE` to `'replay'` and run the test(s).
+
+These tests should work properly without any real connection settings.
+
+## 3 Test Modes
+
+There are 3 representative values of the environment variable `PROMPT_FLOW_TEST_MODE`
+- `live`: Tests run against the real backend, which is the way traditional end-to-end tests do.
+- `record`: Tests run against the real backend, and network traffic will be sanitized (filter sensitive and unnecessary requests/responses) and recorded to local files (recordings).
+- `replay`: There is no real network traffic between SDK/CLI and the backend, tests run against local recordings.
+
+## How to Record a Test
+
+To record a test, don’t forget to clone the full repo and set up the proper test environment following [dev_setup.md](./dev_setup.md):
+1. Prepare some data.
+   * If you have changed/affected tests in __sdk_cli_test__: Copy or rename the file [dev-connections.json.example](../../src/promptflow/dev-connections.json.example) to `connections.json` in the same folder.
+   * If you have changed/affected tests in __sdk_cli_azure_test__: Prepare a config.json, point your `az``  to the proper tenant, subscription, resource group, and workspace.
```

Comment:
I think it's not true for now, tests will honor more on environ or default value, instead of config.json